### PR TITLE
Buf worker fixed

### PR DESCRIPTION
**Pull request by catfact**
redo of #958. 

not sure what happened there, but possible weirdness with nontrivial `std::string` copy constructor usage in `spsc_queue` on ARM. this situation doesn't actually need to be lockfree so just replaced spsc_queue with mutex. (note the FIXME: this means we can replace worker `sleep` with a condvar, but it's a long sleep and its fine for now)

this also brings in small fix for read destination max frame count.

there are quite a few debug statements in the disk worker module. leaving for now, will remove before merge.

### CODE REVIEW

**Comment by catfact** [comment]
i think this is in a pretty good place now.

- tested extensively on device, with the script here: https://github.com/monome/we/pull/26
- added block processing for all operations. interestingly, this makes little if any improvement to read speeds, and massive improvements to write speed. it used to take _90 seconds_ to write a 4-minute stereo file, now it's <2s.

that does lead to the question of whether and how to add scripting events on job completion. there are certainly scenarios where one might want them: for example, if you want to load up some long files and then start playing from arbitrary positions in them, you would want to delay the start of playback until all parts of all files are in memory. but the bookkeeping could get complex on the lua side and maybe it's not worth it, so i'd rather just see how it plays out in the wild and design the API according to need.

there are a couple of minor things i'd like to tidy up, e.g. using a condvar instead of sleep/poll. i'll add these changes here if i get a chance, otherwise they can wait.

File: crone/src/OscInterface.cpp [hunk 1]
```diff
@@ -57,28 +56,24 @@ void OscInterface::init(MixerClient *m, SoftCutClient *sc)
...
     vuPoll = std::make_unique<Poll>("vu");
-    vuPoll->setCallback([](const char* path){
+    vuPoll->setCallback([](const char *path) {
         char l[4];
-	float in0 = mixerClient->getInputPeak(0);
-	float in1 = mixerClient->getInputPeak(1);	
-	float out0 = mixerClient->getOutputPeak(0);	
-	float out1 = mixerClient->getOutputPeak(1);
-
-	l[0] = (uint8_t)(64*Taper::Vu::getPos(in0));
```
Comment (catfact) [inline crone/src/OscInterface.cpp]: sorry, not sure what happened here, will fix

**Comment by tehn** [comment]
that is wild about disk write time!

i hear what you're saying about the completion callbacks and agree about holding off until things get used

i can test on hardware tomorrow, i apologize for the delay.

File: crone/src/OscInterface.cpp [hunk 2]
```diff
@@ -64,6 +64,11 @@ void OscInterface::init(MixerClient *m, SoftCutClient *sc) {
         l[2] = (uint8_t) (64 * mixerClient->getOutputPeak(0));
         l[3] = (uint8_t) (64 * mixerClient->getOutputPeak(1));
 
+        l[0] = (uint8_t)(64*Taper::Vu::getPos(in0));
```
[the remark this answers is not in the document]
Comment (tehn) [reply crone/src/OscInterface.cpp]: i just commented these out to continue testing, but the vu's are wrong of course when i do that

[the remark this answers is not in the document]
Comment (catfact) [reply crone/src/OscInterface.cpp]: i'll fix it its from a bad merge

**Comment by catfact** [comment]
pushed fix for vu meters, sorry about that

@tehn also tested some more. used softcut-studies/files, don't have grid with me today so didn't try MLR. but i'm not seeing any issues.

i'm gonna take a guess that you checked the mono output files with the TAPE page. TAPE doesn't know how to handle mono files and they come out garbled. installed `sox` on the norns and did e.g. `sox ss7-xxxx.wav ss7-xxxx-stereo.wav channels 2` and then that file works fine with TAPE.

**Comment by catfact** [comment]
(BTW i have Tape improvements up next on my todo list.)

**Comment by catfact** [comment]
o

**Pull request merged**